src/libsyntax_pos/hygiene.rs

```diff
@@ -661,13 +647,18 @@ pub struct ExpnInfo {
     /// call_site span would have its own ExpnInfo, with the call_site
     /// pointing to the `foo!` invocation.
     pub call_site: Span,
+    /// The format with which the macro was invoked.
+    pub format: ExpnFormat,
+
+    // --- The part specific to the macro/desugaring definition.
```

Comment:
Could these be in a `struct` for now, even if the actual memory might not be shared?

Reply:
Before changing this I wanted to make a perf run PR making sure that `Lrc` here doesn't actually regress performance (especially with parallel compiler).
It's a tradeoff - spending more memory or going through more locks.

Reply:
`Lrc` wouldn't involve locks though? AFAIK it's `Rc` or `Arc`, and when it's `Arc` you only pay a cost when cloning/dropping a handle (i.e. refcount increment/decrement).
FWIW I was imagining using a `struct` before adding indirection.

Reply:
(Hmm, is there actually a lock when accessing the data behind an `Arc` rather than a reference count, need to check.)

Reply:
Nope! `Deref` on `Arc` is as free as it is on `Rc`.
You only need data synchronization if you have mutation, and then it's separate anyway - e.g. `Rc<RefCell<T>>` and `Arc<Mutex<T>>` .

You can even do `Arc<RefCell<T>>` but that's not `Sync` nor `Send` so it's just a more expensive `Rc<RefCell<T>>`, when incrementing/decrementing the refcount.

Reply:
If `libsyntax_pos` weren't a separate crate, the simplest thing to do would be to just put `Lrc<SyntaxExtension>` here.
Perhaps the hygiene part can be moved back to `libsyntax`, need to check.